### PR TITLE
restore forest deselection from normalize sample weights

### PR DESCRIPTION
**Pull request by ethanglaser**
## Description

Fixes to forest sample weight handling to undo recent forest deselection (https://github.com/uxlfoundation/scikit-learn-intelex/pull/3231). Also addresses older extratrees deselected test.

---

<details>
<summary>Checklist:</summary>

**Completeness and readability**

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with updates and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/doc/sources/contribute.rst) for details)_.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

</details>

### CODE REVIEW

**Comment by ethanglaser** [comment]
http://intel-ci.intel.com/f16fe2e8-4e18-f10f-833f-d4f5ef20c6a0

**Comment by david-cortes-intel** [comment]
@ethanglaser @Alexandr-Solovev This should be done on the oneDAL side, together with other obvious tricks like standardizing 'y', for both CPU and GPU.

File: deselected_tests.yaml [hunk 1]
```diff
@@ -326,7 +326,6 @@ deselected_tests:
   - model_selection/tests/test_classification_threshold.py::test_tuned_threshold_classifier_cv_zeros_sample_weights_equivalence >=1.5
 
   # Deselections for 2025.0
-  - ensemble/tests/test_forest.py::test_importances[ExtraTreesRegressor-squared_error-float64]
```
Comment (david-cortes-intel) [inline deselected_tests.yaml:304]: @ethanglaser This test shouldn't even be getting patched in sklearn1.9 onwards:
https://github.com/scikit-learn/scikit-learn/blob/fa7eddec61a9f336a27e1519161c23b357af138c/sklearn/ensemble/tests/test_forest.py#L339

See the conditions:
https://github.com/uxlfoundation/scikit-learn-intelex/blob/b1d733a6acb74374ffe38c5951de56b180b74d81/doc/sources/algorithms.rst?plain=1#L160

In which job is this failure observed?

Comment (ethanglaser) [reply deselected_tests.yaml:304]: Not sure, this would have been deselected over a year ago

**Comment by ethanglaser** [comment]
oneDAL update is in progress: https://github.com/uxlfoundation/oneDAL/compare/main...ethanglaser:dev/eglaser-forest-fix?expand=1

File: deselected_tests.yaml (hunk 1 above)
Comment (ethanglaser) [reply deselected_tests.yaml:304]: But these are the same test aside from fp type

**Comment by ethanglaser** [comment]
Would merge this after the changes from https://github.com/uxlfoundation/oneDAL/pull/3674 are incorporated into the oneDAL build that github actions fetches

**Comment by ethanglaser** [comment]
CI errors unrelated to this PR. Public CI is update with oneDAL so should be ready.

**Review by david-cortes-intel — APPROVED** [review]
